### PR TITLE
Update jemalloc to version 5.3.0

### PR DESCRIPTION
**Pull request by fwyzard**
This release contains many speed and space optimizations, from micro optimizations on common paths to rework of internal data structures and locking schemes, and many more too detailed to list below. Multiple percent of system level metric improvements were measured in tested production workloads.
The release has gone through large-scale production testing.

See https://github.com/jemalloc/jemalloc/releases/tag/5.3.0 for more details.

Simplify and make more uniform the syntax of the `jemalloc{,-debug,-prof}.spec` files.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
please test for slc7_amd64_gcc10

**Comment by fwyzard** [comment]
please test for el8_amd64_gcc11

**Comment by fwyzard** [comment]
please test for el8_amd64_gcc10

**Comment by fwyzard** [comment]
please test for el8_aarch64_gcc10

**Comment by fwyzard** [comment]
please test for el8_ppc64le_gcc10

**Comment by fwyzard** [comment]
please test for slc7_amd64_gcc10

**Comment by fwyzard** [comment]
@gartung FYI
The [release notes](https://github.com/jemalloc/jemalloc/releases/tag/5.3.0) mention some extra functionality for the flame graphs as well.

**Comment by smuzaffar** [comment]
+externals

**Comment by smuzaffar** [comment]
@fwyzard , PR tests look good, let me know when you are done with your tests

**Comment by fwyzard** [comment]
Running a GPU-enabled HLT menu on one of the new nodes, jemalloc 5.3.0 is **~1% faster** than 5.2.1.

#### jemalloc 5.2.1
```
Running 10 times over 10100 events with 8 jobs, each with 32 threads, 32 streams and 1 GPUs
   621.7 ±   3.5 ev/s
```

#### jemalloc 5.3.0
```
Running 10 times over 10100 events with 8 jobs, each with 32 threads, 32 streams and 1 GPUs
   629.7 ±   6.5 ev/s
```

**Comment by fwyzard** [comment]
I don't know if anyone is interested in measuring the impact on the simulation or the offline reconstruction.

From my point of view, we can merge the PR.

**Comment by fwyzard** [comment]
By the way, just a reminder of what is the performance _without_ jemalloc, using `cmsRunGlibC`:
```
Running 10 times over 10100 events with 8 jobs, each with 32 threads, 32 streams and 1 GPUs
   577.8 ±   2.7 ev/s (based on 9 measurements)
```

File: jemalloc-prof.spec [hunk 1]
```diff
@@ -1,13 +1,14 @@
-Requires: libunwind
 ## INCLUDE jemalloc-common
 ### RPM external jemalloc-prof %{jemalloc_version}
+Requires: libunwind
```
Comment (smuzaffar) [inline jemalloc-prof.spec]: @fwyzard , we need to keep this statement before `## INCLUDE jemalloc-common` to make sure the it is defined before the `%prep` section defined in the `jemalloc-common` . Can you please revert this change ( and may be add a comment why it needs  to be on top)

Comment (fwyzard) [reply jemalloc-prof.spec]: will do

**Comment by fwyzard** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals
